### PR TITLE
[5.1] Fix replacing route default parameters

### PR DESCRIPTION
**Pull request by lindyhopchris**
The fix implemented in #13514 is also required in Laravel 5.1. This pull requests brings the fix from that PR into Laravel 5.1

### CODE REVIEW

**Comment by taylorotwell** [comment]
I will review 5.1 after I merge the other PR.

**Comment by lindyhopchris** [comment]
@taylorotwell ok, np - thought i was helping by submitting this. As the 5.2 fix got merged in May 2016, when can we expect this to land in 5.1?

**Comment by GrahamCampbell** [comment]
The recommended option is for people to upgrade to Laravel 5.4 as soon as they can, rather than staying on an LTS release.

**Comment by lindyhopchris** [comment]
@GrahamCampbell that kind of defeats the point of LTS! According to your support policy, bug fixes are provided for 2 years. With 5.1.0 being tagged in June 2015, that means bug fixes until June 2017.

> Laravel 5.1 is the first release of Laravel to receive long term support. Laravel 5.1 will receive bug fixes for 2 years and security fixes for 3 years. This support window is the largest ever provided for Laravel and provides stability and peace of mind for larger, enterprise clients and customers.

https://laravel.com/docs/5.4/releases#laravel-5.1

**Comment by GrahamCampbell** [comment]
> that kind of defeats the point of LTS

Nah, the point of the LTS (in my opinion) was to get people off 4.x, and then give people the flexibility to do the jump to 5.3 or 5.4 in their own time, without having to worry about major issues or security. There's no guarantee that everything will be fixed on 5.1.

**Comment by GrahamCampbell** [comment]
As far as I'm aware, there's no plan for there ever to be another LTS Laravel release, since too many people are getting stuck on 5.1.

**Comment by GrahamCampbell** [comment]
The other point of 5.1 LTS was to be the last laravel release to support php 5.4, so people stuck on that for whatever reason were still supported.

TLDR: The very latest version of laravel and PHP (currently 5.4.x and 7.1.x) is recommended, rather than say php 5.6 with laravel 5.1.

**Comment by jameswmcnab** [comment]
I would argue that if the LTS release states it receives bug fixes for a specific period then this isn't a promise that should be dropped lightly, regardless of wanting to encourage people to upgrade to a newer version.

Leaving aside LTS releases for a moment the 5.1 branch seems to be having PRs merged consistently so I don't see why this one is any different?

https://github.com/laravel/framework/commits/5.1
